pontoon/base/views.py

```diff
@@ -480,6 +486,28 @@ def delete_translation(request):
     return JsonResponse({"status": True})
 
 
+@utils.require_AJAX
+@login_required(redirect_field_name="", login_url="/403")
+@transaction.atomic
+def delete_comment(request):
+    """Delete given comment."""
+    try:
+        c = request.POST["comment"]
+    except MultiValueDictKeyError as e:
+        return JsonResponse(
+            {"status": False, "message": "Bad Request: {error}".format(error=e)},
+            status=400,
+        )
+
+    comment = get_object_or_404(Comment, pk=c)
+
+    comment.delete()
+
+    log_action("comment:deleted", request.user)
```

Comment:
Why would we want to keep logs about things that don't exist anymore? In the case of a Translation, Entity or Locale, I don't think it makes sense, since we should never request that element ever again, and thus logs become useless.

For users, it's a different scenario, but as far as I know we can't actually properly delete users right now. If we want that to become possible, we should indeed change the on_delete property for the User foreign key to set to null, I guess.

On the comment about this line, this should not work. There should be a validation error when you delete something, because an ActionLog entry needs at least one of `translation` or `entity and locale`.

Reply:
I guess keeping all the activity in the Project/Locale/User timelines permanently would benefit us in monitoring Project/Locale/User activity. OTOH, if I submit a translation by mistake and delete it, I wouldn't mind if traces of that get removed. :)

I wonder what @flodolo and @Pike think. Note that we log the delete action itself.

And to avoid off-topic'ing too much, I also wonder what they think about logging of comment actions (add/remove). Is it good enough if the log action is assigned to a translation (or entity-locale in the case of [team comments](https://docs.google.com/document/d/1W_MEYwtNn-_J0Uiu7gTSwc2B3oux75rjVtjGu-YoFi4/edit#), or do we want to have a comment associated with it?